lib/methods/import.js

```diff
@@ -290,21 +300,42 @@ module.exports = self => {
           reporting.failure();
           continue;
         }
+
+        const isSingleton = self.apos.modules[type] &&
+          self.apos.modules[type].options.singleton === true;
+
+        // Mark singleton and parked docs as "replace" items.
+        // Be sure to always continue the loop to avoid duplicates.
+        // If the parked/singleton doc is not found, it should be processed as a
+        // regular doc.
+        if (isSingleton || parkedId) {
+          if (!aposLocale.endsWith(':draft') || !aposDocId) {
+            continue;
+          }
+          const singletonAposDocId = await self.findExistingSingletonId({
+            type,
+            aposLocale
+          });
+          if (singletonAposDocId) {
+            replaceItems.push({
+              aposDocId,
+              title,
+              type,
+              updatedAt,
+              replaceId: singletonAposDocId
+            });
+            continue;
+          }
+        }
```

Comment:
[Optional]: Maybe we could have used a `Map` here to store singletons list `id -> replaceId` in order to use the same array map above that build the `duplicatedDocs` and having another loop on `replaceId`.
Just an idea here not sure it would really improve code readability / maintainability here

Reply:
I've tried many variants, including trying to extract the whole logic in a function. The real problem is the complex `continue` logic (extremely sensitive to bugs). I prefer to keep it as it is now until we have really good idea how to do it. 
I think the current is clean because its 2 phase logic - 1. either replace or duplicate (no change) set, 1.5. duplicate set only logic (no change), 2. append replace set directly to the result, it's prebuilt.

Reply:
Yes it's good for me.